api/docs/release.dox

```diff
@@ -226,7 +226,10 @@ Further non-compatibility-affecting changes include:
    purpose of preserving register dependencies.
  - Added instr_convert_to_isa_regdeps() API that converts an #instr_t from a real ISA
    (e.g., #DR_ISA_AMD64) to the #DR_ISA_REGDEPS synthetic ISA.
-
+ - Added dr_reg_to_virtual() which takes a regular DR_REG_ register and returns a DR_REG_V
+   virtual register for #DR_ISA_REGDEPS.
+ - Added dr_get_virtual_register_name() which takes a virtual DR_REG_V register (used by
```

Comment:
Long-term if we get multi-arch-in-same-build support we would need the same register printing routine (I guess today's get_register_name()) to work for any ISA.  Should we plan ahead for that and not have an ISA-specific routine like the one added here but instead some mechanism to get which ISA is being asked about.  Probably we won't create _ex versions of every single reg_ routine so presumably this will use the dcontext ISA mode.  So we wouldn't be adding a new API routine.

Reply:
I like the idea of having a single API to get a register's name (regardless of which real/synthetic ISA).
I guess we can use this as an opportunity to use the recent naming convention and have a `dr_get_register_name()` public API, which would deprecate `get_register_name()`.

Do we plan to use any information from `dcontext_t` other than its `isa_mode`?
If not, I think it would be better to have the new `dr_get_register_name()` just take a `dr_isa_mode_t` (and the, already present, `reg_id_t`).
This way a user can decide from where the isa_mode comes from (maybe there are use cases where having it coming from `instr_t` makes more sense?), instead of being "forced" to have a specific ISA set in `dcontext`.

Reply:
If this were the only such API routine I would agree -- but as my comment says we should consider all the other existing register and opnd_t and instr_t routines.  I don't know that we'd make new versions (that's what I meant by `_ex` versions) of every single one to pass in an ISA parameter.  We might have to go with a global (or per-dcontext in managed mode) state instead.

Plus, there is other state like the ISA: the SVE vector length, e.g.  If we took in a parameter it shouldn't just be an ISA but some kind of "ISA state" structure we can append to.

Reply:
Oh, ok, we don't want a new API.
I'll look into "per-dcontext managed mode".
I think if we want to look into having a new "ISA state" structure (rather than just an enum), we should do it in a separate PR and decide there what we want to include.